solver/src/settlement_submission/flashbots_api.rs

```diff
@@ -26,6 +26,20 @@ struct FlashbotGasPrice {
     default: Eip1559,
 }
 
+#[derive(Debug, Deserialize)]
+struct FlashbotStatus {
+    status: String,
+}
+
+#[derive(Deserialize, PartialEq)]
+#[serde(untagged)]
+enum Status {
+    Pending,
+    Failed,
+    Successful,
+    Cancelled,
+}
```

Comment:
nit: Manual serialization should be achievable with `serde`:
```suggestion
#[derive(Debug, Deserialize)]
struct FlashbotStatus {
    status: Status,
}

#[derive(Deserialize, PartialEq)]
enum Status {
    #[serde(rename = "PENDING_BUNDLE")]
    Pending,
    #[serde(rename = "FAILED_BUNDLE")]
    Failed,
    #[serde(rename = "SUCCESSFUL_BUNDLE")]
    Successful,
    #[serde(rename = "CANCEL_BUNDLE_SUCCESSFUL")]
    Cancelled,
}
```

Reply:
Oh, this is nice :) Was searching for this...

Reply:
Fixed.